### PR TITLE
Implement session API

### PR DESCRIPTION
**Pull request by kiootic**
ref #1010 
depends on #1022

### CODE REVIEW

File: pkg/core/auth/session/redis/store.go [hunk 1]
```diff
@@ -73,6 +106,65 @@ func (s *store) Delete(id string) (err error) {
...
+			}
+		}
+
+		if session == nil {
+			// only cleanup expired sessions from the list
+			if expired {
+				// ignore non-critical error
+				_, _ = conn.Do("HDEL", listKey, key)
+			}
+			continue
```
Comment (louischan-oursky) [inline pkg/core/auth/session/redis/store.go]: This `continue` is very subtle and has no effective. Maybe remove it?

File: pkg/auth/handler/session/list.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		"sessions": {
+			"type": "array",
+			"items": { "$ref": "#Session" }
+		}
+	}
+}
+`
+
+/*
+	@Operation POST /session/list - Get current user sessions
```
Comment (louischan-oursky) [inline pkg/auth/handler/session/list.go]: It should be `GET /session`

File: pkg/auth/handler/session/get.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	}
+}
+`
+
+func (p GetRequestPayload) Validate() error {
+	return nil
+}
+
+/*
+	@Operation POST /session/get - Get current user sessions
```
Comment (louischan-oursky) [inline pkg/auth/handler/session/get.go]: It should be `GET /session/<session-id>`

File: pkg/auth/handler/session/get.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+			err = skyerr.NewResourceFetchFailureErr("session", sessionID)
+		}
+		return
+	}
+	if s.UserID != userID {
+		err = skyerr.NewResourceFetchFailureErr("session", sessionID)
+		return
+	}
+
+	resp = authSession.Format(s)
```
Comment (louischan-oursky) [inline pkg/auth/handler/session/get.go]: Wrap the session under the `session` key?

File: pkg/auth/handler/session/revoke.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	}
+}
+`
+
+func (p RevokeRequestPayload) Validate() error {
+	return nil
+}
+
+/*
+	@Operation POST /session/update - Revoke session
```
Comment (louischan-oursky) [inline pkg/auth/handler/session/revoke.go]: It should be `DELETE /session/<session-id>`

File: pkg/auth/handler/session/revoke_test.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+		Convey("should reject current session", func() {
+			payload := RevokeRequestPayload{SessionID: "user-id-1-principal-id-1"}
+			_, err := h.Handle(payload)
+			So(err, ShouldBeError, "InvalidArgument: must not revoke current session")
+			So(sessionProvider.Sessions, ShouldContainKey, "user-id-1-principal-id-1")
+			So(sessionProvider.Sessions, ShouldContainKey, "user-id-1-principal-id-2")
+			So(sessionProvider.Sessions, ShouldContainKey, "user-id-2-principal-id-3")
+		})
+
+		Convey("should non-existing session", func() {
```
Comment (louischan-oursky) [inline pkg/auth/handler/session/revoke_test.go]: should ignore

**Comment by louischan-oursky** [comment]
I removed the comments on more restful API because all APIs are POST.

**Comment by kiootic** [comment]
@louischan-oursky updated, please review

**Pull request merged**